base/gcutils.jl

```diff
@@ -4,17 +4,26 @@
 ==(w::WeakRef, v) = isequal(w.value, v)
 ==(w, v::WeakRef) = isequal(w, v.value)
 
+function _check_mutable(@nospecialize(o)) @noinline
```

Comment:
```suggestion
# Used by `Base.finalizer` to validate mutability of an object being finalized.
function _check_mutable(@nospecialize(o)) @noinline
```
Adding a little context here since the name is pretty broad while the error message is specific to finalizers